src/lsm/segmented_array.zig

```diff
@@ -195,8 +259,10 @@ fn SegmentedArrayType(
 
             const a = cursor.node;
             const a_pointer = array.nodes[a].?;
+            const a_count = array.count(a);
```

Comment:
We tried to avoid caching counts like this within scopes in SegmentedArray because it's easy for them to get out of sync, especially when there's significant distance in terms of lines of code between where they are cached and where they are used, i.e. a spatial form of TOCTOU.